lib/public/init.js

```diff
@@ -27,6 +27,16 @@ module.exports = function (browserSync, name, pjson) {
 
         args.config.version = pjson.version;
 
+        /**
+         * Preserve the httpModule property's functions.
+         * the http2 module exports an object of functions and the merge function seems
+         * to want to destroy that, but if the base object is a function it seems fine
+         * TODO: find a better or more generic way to handle this
+         */
+        if(args.config.httpModule && !_.isFunction(args.config.httpModule)) {
+            args.config.httpModule = Object.assign(function() {}, args.config.httpModule);
+        }
```

Comment:
passing in the *actual* module I think is problematic, for this & other reasons.

also because we want this to work on the CLI, it's easier to just allow users to give a string that we pass to nodes `require()` see https://github.com/BrowserSync/browser-sync/commit/8ba178e7505078a0a6e533ea57596a6acf62f565